src/main/java/com/google/sps/servlets/AddSaleCardServlet.java

```diff
@@ -99,48 +107,77 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
 
       DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
 
-      // TODO: Create vendor key using the Auth ID
-      // Key vendorKey = KeyFactory.createKey("Vendor", (String) user.authID);
-
-      // Generate Key when datastore put
-      Entity saleCard = new Entity("SaleCard"/*, vendorKey*/);
-      datastore.put(saleCard);      
-
-      // TODO: Remove from datastore existing saleCard, picture and location data
+      Key vendorKey = KeyFactory.createKey("Vendor", userService.getCurrentUser().getUserId());
+      Entity vendorEntity;
+      try {
+        vendorEntity = datastore.get(vendorKey);
+      } catch (EntityNotFoundException e) {
+        System.out.println(e);
+        response.sendError(HttpServletResponse.SC_BAD_REQUEST);
+        return;
+      }
+      Vendor vendorObject = new Vendor(vendorEntity);
+
+      Entity saleCard, picture, locationData;
+      // If the vendor doesn't have a saleCard, create completely new Entities (datastore.put(...) to generate Key)
+      // Else, use the already existing Entities
+      if (vendorObject.getSaleCard() == null) {
```

Comment:
Nit: I'm not sure how this works in Java, but in C++ and JavaScript, you can do:

if (!vendorObject.getSaleCard()) {...}

Reply:
Yep, it does not work, the compiler throws an error. 😞